### DISCUSSION TITLE
v1.3.1

### DISCUSSION THREAD

**Discussion by nperovic** [nperovic/ColorButton.ahk#6] [Announcements]
![image](https://github.com/nperovic/ColorButton.ahk/assets/122501303/56747bad-9ab7-47b9-b1f4-1906448be2a5)

## New Feature
- The **border width** of the button is now customisable with `ShowBorder` parameter or `ShowBorder` property.

### `ShowBorder`
```ts
/**
 * @prop {Integer} ShowBorder
 * Border preference. (To set the border preference, you must have used `SetColor()`, `SetBackColor()`, or `BackColor` to set the background colour at least once beforehand.)
 * - `n` : The higher the value, the thicker the button's border when focused.
 * - `1` : Highlight when focused.
 * - `0` : No border displayed.
 * - `-1`: Border always visible.
 * - `-n`: The lower the value, the thicker the button's border when always visible.
 */
```

**Full Changelog**: https://github.com/nperovic/ColorButton.ahk/compare/v1.3.0...v1.3.1

<hr /><em>This discussion was created from the release <a href='https://github.com/nperovic/ColorButton.ahk/releases/tag/v1.3.1'>v1.3.1</a>.</em>

**Comment by ejbytes** [comment]
BUG lin 368 no __hbrush variable declaired


I used this in a new project today. 
When I X'd out of the GUI "window", not a button, but the X on the window itself. I got an error.

Paste Error:

Error: This value of type "Gui.Button" has no property named "__hbrush".

	---- ColorButton.ahk
	376: {
▶	376: Return  DeleteObject(this.__hbrush)
	376: }
	376: this.Gui.OnEvent("Close", (*) => DeleteObject(this.__hbrush))

End Paste.


To "fix" it, I declared the variable and instaticated it on line 368
this.__hbrush := 0


NOTE:
this.__hbrush := 0 was necessary, because when I tried this.__hbrush := "" 
I got another error, "expected a number but got an empty string"


I am using PC, Win 10 latest stable release AHK V2.0.19  STABLE release.
